doc/design/apex-domain-netlify.md

```diff
@@ -0,0 +1,38 @@
+# Serving the GUI at dandiarchive.org with Netlify
+Currently, the GUI is served at gui.dandiarchive.org, while the apex domain dandiarchive.org is controlled by the redirector. The redirector performs two main tasks
+
+1. Redirect requests from dandiarchive.org to gui.dandiarchive.org
+2. Serve JSON information about deployed services (e.g. urls for the API, web ui, and jupyterhub) at dandiarchive.org/server-info.
+
+We've decided that our UI is stable enough to warrant serving it directly at dandiarchive.org, leaving just the task of serving a JSON file containing deployment information. Netlify seems capable of performing this task, and this document outlines the proposal to do so.
+
+## Serving the Deployment JSON file
+
+Allowing Netlify to serve this deployment info can be broken down into two sub-tasks
+
+1. Serving the JSON file at sub path of dandiarchive.org without interfering with the GUI
+2. Obtaning the statically coded JSON file at or before client build time, so that Netlify may serve it
+
+## Serving the JSON file
+
+This is easily accomplished with Netlify [redirects](https://docs.netlify.com/routing/redirects/). The additional code this would amount to would resemble the following
+
+```toml
+# netlify.toml
+
+[[redirects]]
+from = "/server-info"
+to = "/server-info.json"
+status = 200
+```
+
+## Obtaining the static JSON file
+
+Since Netlify is merely serving content for us, we need a way to include this JSON file as a local file, either as a part of, or a prerequisite to the build process. Rather than hardcoding this file in the source code, where it must be regularly updated, the API `/api/info` endpoint could be augmented to include this information as well. The Netlify build process would then simply be updated to download this file and serve it as shown above. The fetching and downloading of this file can be done through Nelify [Build Plugins](https://docs.netlify.com/configure-builds/build-plugins/).
```

Comment:
If we're already serving this information at `/api/info`, is there a specific reason we need to also serve this file from the frontend? (attn: @satra, @yarikoptic)

Reply:
Obviously Satra and Yarik can answer, but I believe it's to have an index into the api (as well as other related services) attached to the main site itself. For other instances it might not be as simple as just `api.<instance>`.

Reply:
yup. we wanted a single point of entry into the services and that should be from the domain endpoint.